sriovnet.go

```diff
@@ -407,6 +409,24 @@ func GetVfNetdevName(handle *PfNetdevHandle, vf *VfObj) string {
 	return vfNetdevNameFromParent(handle.PfNetdevName, vf.Index)
 }
 
+// Finds the VF index of a port, if applicable
+func GetVfIndexFromNetDevice(name string) (int, error) {
```

Comment:
An alternative implementation which is not related to switchdev is to get VF PCI address from net device and use `GetVfIndexByPciAddress`
(essentially reading link of /sys/class/net/<netdev>/device if exists or just using `GetPciFromNetDevice` which you defined.)

https://github.com/Mellanox/sriovnet/blob/e360783a88a929a3100628cb7ce27ed029c276f9/sriovnet.go#L412

Reply:
> Hey there @grivet !
> 
> where are you planning to use these methods if i may ask ?

Hey !

I am using those methods on the DPU, to gather VF representor details before plugging those ports in DPDK.
They are meant to execute on DPU-side only.

Ack on the switchdev comment, will move it there.

For the alternative implementation, that was my first attempt, but it won't work on the DPU side.
On a BF2 system, with 8 VF allocated, here is what's exposed in the sysfs:

x86 side:
```bash
[root@k8s-worker1 ~]# ls -l /sys/class/net/p0/device/virtfn*
lrwxrwxrwx 1 root root 0 Jul 11 16:18 /sys/class/net/p0/device/virtfn0 -> ../0000:05:00.3
lrwxrwxrwx 1 root root 0 Jul 11 16:18 /sys/class/net/p0/device/virtfn1 -> ../0000:05:00.4
lrwxrwxrwx 1 root root 0 Jul 11 16:18 /sys/class/net/p0/device/virtfn2 -> ../0000:05:00.5
lrwxrwxrwx 1 root root 0 Jul 11 16:18 /sys/class/net/p0/device/virtfn3 -> ../0000:05:00.6
lrwxrwxrwx 1 root root 0 Jul 11 16:18 /sys/class/net/p0/device/virtfn4 -> ../0000:05:00.7
lrwxrwxrwx 1 root root 0 Jul 11 16:18 /sys/class/net/p0/device/virtfn5 -> ../0000:05:01.0
lrwxrwxrwx 1 root root 0 Jul 11 16:18 /sys/class/net/p0/device/virtfn6 -> ../0000:05:01.1
lrwxrwxrwx 1 root root 0 Jul 11 16:18 /sys/class/net/p0/device/virtfn7 -> ../0000:05:01.2
```

DPU side:
```bash
root@k8s-worker1-bf:~# ls -l /sys/class/net/p0/device/virtfn*
ls: cannot access '/sys/class/net/p0/device/virtfn*': No such file or directory
root@k8s-worker1-bf:~# cat /sys/class/net/pf0vf7/phys_port_name
pf0vf7
```

So, to get the VF index from the DPU side, we cannot rely on scouring the PCI bus sysfs and find the proper device, which is why I had to resort matching on the content of phys_port_name.
Unfortunately, this same operation cannot work on the x86 side.